api/lib/domain/models/OrganizationInvitedUser.js

```diff
@@ -0,0 +1,41 @@
+const { AlreadyExistingMembershipError } = require('../../domain/errors');
+const { NotFoundError } = require('../errors');
+const { roles } = require('./Membership');
+const OrganizationInvitation = require('./OrganizationInvitation');
+
+class OrganizationInvitedUser {
+  constructor({ userId, invitation, currentRole, organizationHasMemberships, currentMembershipId, status } = {}) {
+    this.userId = userId;
+    this.invitation = invitation;
+    this.currentRole = currentRole;
+    this.organizationHasMemberships = organizationHasMemberships;
+    this.currentMembershipId = currentMembershipId;
+    this.status = status;
+  }
+
+  acceptInvitation({ code }) {
+    if (code !== this.invitation.code) {
+      throw new NotFoundError(`Not found organization-invitation for ID ${this.invitation.id} and code ${code}`);
+    }
+
+    if (this.currentRole && !this.invitation.role) {
```

Comment:
**suggestion :** SI je supprime le `&& !this.invitation.role` tous les tests passent